src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleMetadata.cs

```diff
@@ -18,14 +18,19 @@ public static partial class JsonSerializer
             ref Utf8JsonReader reader,
```

Comment:
How's the test coverage for `ResolveMetadata`?

Reply:
It does have some gaps - Line 84.35%, Branch 92.64%. Addressing this is out of scope for this PR - we should address this as part of https://github.com/dotnet/runtime/issues/32341 & likely https://github.com/dotnet/runtime/issues/35517.